driver-core/src/main/com/mongodb/internal/connection/MultiServerCluster.java

```diff
@@ -82,19 +86,45 @@ public MultiServerCluster(final ClusterId clusterId, final ClusterSettings setti
             LOGGER.info(format("Cluster created with settings %s", settings.getShortDescription()));
         }
 
+        if (settings.getSrvHost() != null) {
```

Comment:
Resolution is being done in a thread now.  Once we add polling, this will become more like DefaultServerMonitor in that it will loop without exiting.